main-command/src/main/scala/sbt/internal/CommandChannel.scala

```diff
@@ -20,31 +21,59 @@ import sbt.protocol.EventMessage
  */
 abstract class CommandChannel {
   private val commandQueue: ConcurrentLinkedQueue[Exec] = new ConcurrentLinkedQueue()
-  private val registered: java.util.Set[java.util.Queue[CommandChannel]] = new java.util.HashSet
-  private[sbt] final def register(queue: java.util.Queue[CommandChannel]): Unit = {
-    registered.add(queue)
-    ()
+  private val registered: java.util.Set[java.util.Queue[Exec]] = new java.util.HashSet
+  private val maintenance: java.util.Set[java.util.Queue[MaintenanceTask]] = new java.util.HashSet
+  private[sbt] final def register(
+      queue: java.util.Queue[Exec],
+      maintenanceQueue: java.util.Queue[MaintenanceTask]
+  ): Unit =
```

Comment:
What's a maintenanceQueue?

Reply:
Just reread the commit message:

> Certain commands such as `exit` should be evaluated immediately. To make
this work, we add the concept of a MaintenanceTask. The CommandExchange
has a background thread that reads MaintenanceTasks and evaluates them
on demand. This allows maintenance tasks to be evaluated even when sbt
is evaluating an exec. If it weren't done this way, when the user typed
exit while a different remote connection was running a command, they
wouldn't be able to exit until the command completed.

What do you think about calling this `FastTrackTask` and `FastTrackQueue` instead? Maintenance sounds like it runs at the end as a clean-up job.

Reply:
Sure: https://github.com/sbt/sbt/pull/5620/commits/ba0d97c9ac8d463502050900a8396785e8d2465c